composer/trainer/trainer.py

```diff
@@ -97,37 +128,85 @@ class contains metrics relevant to the specific dataset. Set to ``None`` for no
-        save_folder (str, optional): Folder path to save checkpoints, relative to the run directory.
-            Set to ``None`` to not save checkpoints. (default: ``None``)
-        save_interval (str or int): How often to save checkpoints. For example, set to "1ep" to save checkpoints
+
+            .. note:: In order to get reproducible results, call the
+                :func:`.seed_all` function at the start of your script with the seed
+                passed to the trainer. This will ensure any initialization done before the trainer init
+                (ex. model weight initialization) also uses the provided seed.
+
+            .. seealso:: :mod:`composer.utils.reproducibility` for more details on reproducibility.
+        deterministic_mode (bool, optional): Run the model deterministically. (default: ``False``)
+
+            .. note:: This is an experimental feature. Performance degradations expected. Certain Torch modules may
+                not have deterministic implementations, which will result in a crash.
+
+            .. seealso:: :mod:`composer.utils.reproducibility` for more details on reproducibility.
+        loggers (Sequence[LoggerCallback], optional): The destinations to log training information to.
+            If ``None``, will be set to ``[TQDMLogger()]``. (default: ``None``)
+
+            .. seealso:: :mod:`composer.loggers` for the different loggers built into Composer.
+        callbacks (Sequence[Callback], optional): The callbacks to run during training. If ``None``,
+            then no callbacks will be run. (default: ``None``).
+
+            .. seealso:: :mod:`composer.callbacks` for the different callbacks built into Composer.
+        load_path (str, optional):  The template path to an existing checkpoint file.
+            It can be a path to a file on local disk, a URL, or if ``load_object_store`` is set, the object name
+            for a checkpoint in a cloud bucket.
+
+            When using `Deepspeed ZeRO <https://www.deepspeed.ai/tutorials/zero/>`_, saved checkpoints are
+            sharded rank. To load deepspeed checkpoints, specify ``{RANK}`` in this ``path``
+            parameter, and the ``RANK`` variable will be substituted with the global rank, thus allowing the correct
+            checkpoints to be loaded per-rank.
+
+            For example, suppose that checkpoints are stored in the following structure:
+
+            .. code-block::
+
+                my_model/rank_0/ep1.tar
+                my_model/rank_1/ep1.tar
+                my_model/rank_2/ep1.tar
+                ...
+
+            Then, ``path`` should be set to ``my_model/rank_{RANK}/ep1.tar``, and all ranks will load the correct
+            data.
+
+            If ``None`` then no checkpoint will be loaded. (default: ``None``)
+        load_object_store (ObjectStoreProvider, optional): If the ``load_path`` is in an object store
```

Comment:
Can you add an example of loading a checkpoint from an object store? I feel that it is advanced feature to have a parameterized path combined with retrieving a checkpoint from a bucket.